Sources/DistributedActors/Pattern/WorkerPool.swift

```diff
@@ -12,18 +12,28 @@
 //
 //===----------------------------------------------------------------------===//
 
+import Distributed
 import Logging
 
-/// A `WorkerPool` represents a pool of actors that are all equally qualified to handle incoming messages.
+public protocol DistributedWorker: DistributedActor {
+    associatedtype WorkItem: Codable
+    associatedtype WorkResult: Codable
+    
+    distributed func submit(work: WorkItem) async throws -> WorkResult
+}
+
+/// A `WorkerPool` represents a pool of actors that are all equally qualified to handle incoming work items.
 ///
 /// Pool members may be local or remote, // TODO: and there should be ways to say "prefer local or something".
 ///
 /// A pool can populate its member list using the `Receptionist` mechanism, and thus allows members to join and leave
 /// dynamically, e.g. if a node joins or removes itself from the cluster.
 ///
 // TODO: A pool can be configured to terminate itself when any of its workers terminate or attempt to spawn replacements.
-public struct WorkerPool<Message: ActorMessage> {
-    typealias Ref = WorkerPoolRef<Message>
+public distributed actor WorkerPool<Worker: DistributedWorker>: DistributedWorker, LifecycleWatch, CustomStringConvertible where Worker.ActorSystem == ClusterSystem {
```

Comment:
If we'd add the where clause to the DistributedWorker protocol it could likely go away from here 🤔

Reply:
Yeah that's what I had to begin with but the compiler wasn't happy.

Tried this just now:

```
/code/Sources/DistributedActors/Pattern/WorkerPool.swift:83:60: error: generic parameter 'Guest' could not be inferred
                for await worker in await self.actorSystem.receptionist.subscribe(to: key) {
                                                           ^
/code/Sources/DistributedActors/Cluster/Reception/OperationLogDistributedReceptionist.swift:346:29: note: in call to function 'subscribe(to:)'
    public nonisolated func subscribe<Guest>(
                            ^
/code/Sources/DistributedActors/Pattern/WorkerPool.swift:97:17: error: instance method 'watchTermination(of:whenTerminated:file:line:)' requires the types 'Worker.ActorSystem' and 'ClusterSystem' be equivalent
                watchTermination(of: worker) { self.onWorkerTerminated(id: $0) }
```

Reply:
I see... I'll have a look if that's maybe compiler bug, thanks!